### PR TITLE
Add document for language syntax

### PR DESCRIPTION
**Pull request by scottoasis**
## Why A Premature Pull Request?
1. We can collaborate;
2. By publishing your output as early as possible, you can get as many chances as possible for others to review it (even if it's neither complete nor perfect).

### CODE REVIEW

**Comment by scottoasis** [comment]
@yorkie: May I ask you to review this outline structure?

File: doc/syntax.org [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#+TITLE:   Markup Syntax Guide for Pancacke(d)
+#+AUTHOR:  Scott Wang
+#+STARTUP: align indent
+#+OPTIONS: H:5
+
+
+* Overview
```
Comment (yorkie) [inline doc/syntax.org:7]: Currently I assume the top level headline _MUST_ be in uppercase, should we keep this?

Comment (scottoasis) [inline doc/syntax.org]: In Org Mode or in _Pancake-Flavored_ Org Mode? For former ones, not necessary.

Comment (scottoasis) [inline doc/syntax.org:7]: In Org Mode or in _Pancake-Flavored_ Org Mode? For former one, not necessary.

Comment (yorkie) [inline doc/syntax.org:7]: Should be in _Pancake-Flavored Org Mode_, the source is at https://github.com/weflex/pancake/blob/master/lib/pancake.js#L152

Comment (scottoasis) [inline doc/syntax.org:7]: Then we can declare it in the `Headlines` section or `Keywords and Reserved Words` section. This document (the document file this PR modifies) itself is the language specification.

Comment (yorkie) [inline doc/syntax.org:7]: I mean do we need apply the rule in this document itself?

Comment (scottoasis) [inline doc/syntax.org:7]: This is just a plain document for the language, so not this one. Is that work for you?

Comment (yorkie) [inline doc/syntax.org:7]: Sure :-) Now waiting for the other parts of this language.

Comment (scottoasis) [inline doc/syntax.org:7]: You can submit what you want to specify with this language as well :) that's what I meant by _collaborate_.

Comment (yorkie) [inline doc/syntax.org:7]: What I strongly recommend and want is the meta type, which would make it possible do define functions which can be used at `embeds` and `models` sections.

For example, we can absolutely implement the `hasMany`, `hasOne`, `Enum` and `Range` by define them as meta types.

**Pull request merged**